adapters/sdk.go

```diff
@@ -208,11 +208,41 @@ func (b *SDKAdapter) GetBlockByID(
 			Height:    flowBlock.Header.Height,
 			Timestamp: flowBlock.Header.Timestamp,
 		},
-		BlockPayload: sdk.BlockPayload{},
+		BlockPayload: convertBlockPayload(flowBlock.Payload),
 	}
 	return &block, sdk.BlockStatusSealed, nil
 }
 
+func convertBlockPayload(payload *flowgo.Payload) sdk.BlockPayload {
+	var seals []*sdk.BlockSeal
+	sealCount := len(payload.Seals)
+	if sealCount > 0 {
+		seals = make([]*sdk.BlockSeal, 0, sealCount)
+		for _, seal := range payload.Seals {
+			seals = append(seals, &sdk.BlockSeal{
+				BlockID:            sdk.Identifier(seal.BlockID),
+				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
+			})
+		}
+	}
```

Comment:
a bit cleaner (can do same bellow).
```suggestion
	seals := make(make([]*sdk.BlockSeal, len(payload.Seals))
	for i, seal := range payload.Seals {
			seals[i] = &sdk.BlockSeal{
				BlockID:            sdk.Identifier(seal.BlockID),
				ExecutionReceiptID: sdk.Identifier(seal.ResultID),
			}
		}
	}
```

Reply:
See https://github.com/onflow/flow-emulator/pull/704#discussion_r1667773134. The current code doesn't allocate unless necessary

Reply:
I believe if capacity or length is set to 0, nothing should be allocated either: 

https://github.com/golang/go/blob/6d89b38ed86e0bfa0ddaba08dc4071e6bb300eea/src/runtime/slice.go#L116

https://github.com/golang/go/blob/6d89b38ed86e0bfa0ddaba08dc4071e6bb300eea/src/runtime/malloc.go#L988-L990

But yeah, It's probably still clearer to explicitly check for length in the application code.